apollo-router/src/services/subgraph_service.rs

```diff
@@ -381,6 +392,23 @@ mod tests {
         }
     }
 
+    // starts a local server emulating a subgraph returning status code 401
+    async fn emulate_subgraph_unauthorized(socket_addr: SocketAddr) {
+        async fn handle(_request: http::Request<Body>) -> Result<http::Response<Body>, Infallible> {
+            Ok(http::Response::builder()
+                .header(CONTENT_TYPE, "text/html")
+                .status(StatusCode::UNAUTHORIZED)
+                .body(r#""#.into())
+                .unwrap())
+        }
+
+        let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(handle)) });
+        let server = Server::bind(&socket_addr).serve(make_svc);
+        if let Err(e) = server.await {
```

Comment:
Could we panic instead ? If it's for a test it's better to directly panic

Reply:
I just copied the precedence set in the other `emulate_*` functions. 
This is my first time writing any Rust code so I'll leave this judgement call up to you.

Reply:
Yes I know it's our mistake, we should change the other ones. We will... one day :)